src/OpenTelemetry.Abstractions/Context/Propagation/StringExtensions.cs

```diff
@@ -0,0 +1,111 @@
+﻿// <copyright file="StringExtensions.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+namespace OpenTelemetry.Context.Propagation
+{
+    using System;
+    using System.Collections.Generic;
+    using OpenTelemetry.Trace;
+
+    /// <summary>
+    /// Extension methods to extract Tracestate from string.
+    /// </summary>
+    public static class StringExtensions
+    {
+        /// <summary>
+        /// Extracts <see cref="Tracestate"/> from the given string and sets it on provided <see cref="Tracestate.TracestateBuilder"/>.
+        /// </summary>
+        /// <param name="tracestateString">String with comma separated tracestate key value pairs.</param>
+        /// <param name="tracestateBuilder"><see cref="Tracestate.TracestateBuilder"/> to set tracestate pairs on.</param>
+        /// <returns>True if string was parsed successfully and tracestate was recognized, false otherwise.</returns>
+        public static bool TryExtractTracestate(this string tracestateString, Tracestate.TracestateBuilder tracestateBuilder)
```

Comment:
is it needed to be an extension method on string? Also, I see it is used in both project. But do we really need to make it a part of API. Can be better to reuse the code

Reply:
So it was not trivial. 
I didn't want to copy-paste, but I wanted to add the same source code to 2 projects: Abstractions and SDK.

Since it's `internal` and internals are visible to tests, tests found this class in 2 dlls. I've played with aliases and they would complicate things beyond acceptable.

I decided to separate abstractions and sdk tests to fix this issue.